Automattic/wp-calypso
# account: remove lib/username and use internal methods instead

#### Changes proposed in this Pull Request

* Remove `lib/username` and use internal component state & methods instead

#### Testing instructions

Note: You have to use a non-a8c with a verified email address to follow the testing instructions 

* Go to `me/account` and attempt to change your username
* Make sure validation runs (inspect network requests) and shows appropriate error messages (see code for possible validation failures)
* Now enter a new valid username and hit _Save username_
* The page should reload (if 2fa auth is activated on that account you should be prompted to enter a verification code)

Variants: tick the box that you want a new blog or change an existing blog to match your new username

Fixes #24128
